### ISSUE TITLE
Call to undefined method Illuminate\Mail\Transport\LogTransport::setPingThreshold()

### ISSUE THREAD

**Issue by nicholaszuccarelli** [juhasev/laravel-ses#19]
Recently upgraded to Laravel 9 (now using the v2 branch).
Attempting to enable email tracking when sending an email 
```
$handler = SesMail::enableAllTracking();
$handler->send(...);

```
is throwing the error:
```
[2023-03-23 20:07:51] local.ERROR: Call to undefined method Illuminate\Mail\Transport\LogTransport::setPingThreshold()  

[2023-03-23 20:07:51] local.ERROR: 77  

[2023-03-23 20:07:51] local.ERROR: /var/www/html/vendor/juhasev/laravel-ses/src/LaravelSesServiceProvider.php  

[2023-03-23 20:07:51] local.ERROR: Error: Call to undefined method Illuminate\Mail\Transport\LogTransport::setPingThreshold() in /var/www/html/vendor/juhasev/laravel-ses/src/LaravelSesServiceProvider.php:77
```

My ENV is using the `log` mail driver.

Backtracking through LaravelSesServiceProvider, app('mailer') exists, as does ->getSymfonyTransport();
As the error says, it fails when attempting `setPingThreshold()`

**Comment by leMaur** [comment]
Hi @nicholaszuccarelli,

If you use Laravel 9.x you can install the latest version available v3.0.4.
You can upgrade your project by running `composer require juhasev/laravel-ses:3.0.4`.

Please don't hesitate to let me know if you encounter any issues.

–Maurizio

**Comment by nicholaszuccarelli** [comment]
Thanks for that. Just noticed that you removed the comment about using the v2 branch.
I'll leave this thread open and test it on Monday to see if it's resolved :)

**Comment by nicholaszuccarelli** [comment]
Hey again @leMaur
I have pulled v3.0.4 and I am still encountering the same error and line number (77)

**Comment by nicholaszuccarelli** [comment]
To add further details. ->getSymfonyTransport() is returning an Illuminate\Mail\Transport\LogTransport instance which does not contain the setPingThreshold() method (I am using the `log` driver).
Perhaps the service provider needs to check if the setPingThreshold() method exists first?

The same is for line 84 where `setRestartThreshold()` doesn't exist either.

**Comment by nicholaszuccarelli** [comment]
Another update. I attempted to change my MAIL_MAILER to SES.
Also not working. It is returning a Illuminate\Mail\Transport\SesTransport instance which also doesn't contain the `setPingThreshold` or `setRestartThreshold` methods

In artisan tinker, I attempted to do `app('SesMailer')` and its also returning the same error.

---

The `setPingThreshold` method only exists for Symfony\Component\Mailer\Transport\Smtp\SmtpTransport

The `->getSymfonyTransport()` method only returns a `TransfportInterface` which can have a `send()` and `__toString()` method

**Comment by juhasev** [comment]
Ni Nicholas,

Please read the following:

https://laravel.com/docs/9.x/upgrade#symfony-mailer

Sounds like you may have missed some of the upgrade steps. We run the latest version with the latest Laravel 9.x without issues.

-- Juha

**Comment by nicholaszuccarelli** [comment]
Hi Juhasev.

I have followed the steps, and even attempted it on a blank project. I am still receiving the same error on a blank project.

The only time I am not getting any errors is when the MAIL_MAILER = smtp
If I set my MAIL_MAILER to ses, log (or any other driver), I am receiving the error

As I mentioned in my previous posts, those methods do not exist for any other drivers except the Smtp driver.
